xdsl/dialects/x86/ops.py

```diff
@@ -1026,6 +1034,8 @@ class RS_AddOp(RS_Operation[GeneralRegisterType, GeneralRegisterType]):
 
     name = "x86.rs.add"
 
+    traits = traits_def(Pure(), RS_AddOpHasCanonicalizationPatterns())
```

Comment:
Is this definitely pure? Doesn't it modify the input register `r`, or do you enforce that `r` isn't used again after this op?